silq/instrument_interfaces/keysight/E8267D_interface.py

```diff
@@ -264,25 +267,60 @@ def get_additional_pulses(self, connections) -> List[Pulse]:
             else:  # value away from center
                 frequency_carrier = int(round((min_frequency + max_frequency) / 2))
                 frequency_carrier += self.frequency_carrier_choice()
-            self.frequency(frequency_carrier)
 
+            settings['frequency'] = frequency_carrier
+        else:
+            settings['frequency'] = self.frequency()
+
+        # Determine frequency deviation
         if not self.fix_frequency_deviation():
-            self.frequency_deviation(
-                int(round(max([max_frequency - self.frequency(),
-                               self.frequency() - min_frequency]))))
+            frequency_deviation = int(round(max(
+                [max_frequency - self.frequency(), self.frequency() - min_frequency]
+            )))
+            settings['frequency_deviation'] = frequency_deviation
+        else:
+            settings['frequency_deviation'] = self.frequency_deviation()
```

Comment:
Why are you determining frequency deviation based on the current self.frequency(), but not with respect to the self.frequency(frequency_carrier)?